src/main/java/com/google/devtools/build/lib/sandbox/AbstractSandboxSpawnRunner.java

```diff
@@ -238,23 +239,36 @@ private boolean wasTimeout(Duration timeout, Duration wallTime) {
           testTmpdir,
           "Cannot resolve symlinks in TEST_TMPDIR because it doesn't exist: \"%s\"");
     }
-    addWritablePath(
-        sandboxExecRoot,
-        writablePaths,
-        // As of 2018-01-09:
-        // - every caller of `getWritableDirs` passes a LocalEnvProvider-processed environment as
-        //   `env`, and in every case that's either PosixLocalEnvProvider or XcodeLocalEnvProvider,
-        //   therefore `env` surely has an entry for TMPDIR
-        // - Bazel-on-Windows does not yet support sandboxing, so we don't need to add env[TMP] and
-        //   env[TEMP] as writable paths.
-        Preconditions.checkNotNull(env.get("TMPDIR")),
-        "Cannot resolve symlinks in TMPDIR because it doesn't exist: \"%s\"");
+    // As of 2019-07-08:
+    // - every caller of `getWritableDirs` passes a LocalEnvProvider-processed environment as
+    //   `env`, therefore `env` surely has an entry for TMPDIR on Unix and TEMP/TMP on Windows.
+    if (OS.getCurrent() == OS.WINDOWS) {
+      addWritablePath(
+          sandboxExecRoot,
+          writablePaths,
+          Preconditions.checkNotNull(env.get("TEMP")),
+          "Cannot resolve symlinks in TEMP because it doesn't exist: \"%s\"");
+      addWritablePath(
+          sandboxExecRoot,
+          writablePaths,
+          Preconditions.checkNotNull(env.get("TMP")),
+          "Cannot resolve symlinks in TMP because it doesn't exist: \"%s\"");
+    } else {
+      addWritablePath(
+          sandboxExecRoot,
+          writablePaths,
+          Preconditions.checkNotNull(env.get("TMPDIR")),
+          "Cannot resolve symlinks in TMPDIR because it doesn't exist: \"%s\"");
+    }
 
     FileSystem fileSystem = sandboxExecRoot.getFileSystem();
     for (String writablePath : sandboxOptions.sandboxWritablePath) {
       Path path = fileSystem.getPath(writablePath);
       writablePaths.add(path);
-      writablePaths.add(path.resolveSymbolicLinks());
+      // TODO(laszlocsomor) Remove if guard when path.resolveSymbolicLinks supports non-symlink
+      if (path.isSymbolicLink()) {
```

Comment:
> The PR seems to break an internal test but I'm not sure why. Investigating.

This new if-condition, and the one I commented on below, are responsible.

This workaround will do for now:

```diff
-	      if (path.isSymbolicLink()) {
+	      // TODO(laszlocsomor): figure out why OS.getCurrent() != OS.WINDOWS is required, and remove it
+	      if (OS.getCurrent() != OS.WINDOWS || path.isSymbolicLink()) {
```